### ISSUE TITLE
Temporary Access for Collaborators

### ISSUE THREAD

**Issue by terowz** [github/hub#2984]
**The problem I'm trying to solve:**

There are many times when a group of contractors or nonemployees need access to one of our Github repos. It would be great if Github provided a way to give temporary collaborator access. After end date the user is removed from the org/repo.

**How I imagine hub could expose this functionality:**

1. When adding a collab to a repo, some type of date picker would show up, with defaults:

- 1 day
- 1 week
- 1 month
- 1 year
- -- choose time --

When the date comes the user would lose their access, no matter the level R/W.

2. This functionality should be available in the API if it's considered.

**Comment by rmolina2** [comment]
Yes this world be great. Just for someone to help on a project we have to purchase a new seat license when they will only need access for a few months.
